File: src/main/java/szewek/fl/proxy/FLProxyClient.java
package szewek.fl.proxy;

import net.minecraft.client.Minecraft;
import net.minecraft.client.renderer.ItemModelMesher;
import net.minecraft.client.renderer.block.model.ModelResourceLocation;
import net.minecraft.item.Item;
import net.minecraftforge.fml.relauncher.Side;
import net.minecraftforge.fml.relauncher.SideOnly;
import szewek.fl.util.PreRegister;

import java.util.ArrayList;
import java.util.List;
import java.util.Set;

@SideOnly(Side.CLIENT)
public class FLProxyClient extends FLProxy {
	private final List<PreRegister> prl = new ArrayList<>();
	@Override
	public void addPreRegister(PreRegister pr) {
		prl.add(pr);
	}

	public void init() {
		final ItemModelMesher imm = Minecraft.getMinecraft().getRenderItem().getItemModelMesher();
		for (PreRegister pr : prl) {
			Set<Item> iset = pr.getItems();
			for (Item i : iset) {
				ModelResourceLocation mrl = new ModelResourceLocation(i.getRegistryName(), "inventory");
				imm.register(i, 0, mrl);
			}
		}
		prl.clear();
	}
}

File: src/main/java/szewek/fl/proxy/FLProxyClient.kt
package szewek.fl.proxy

import net.minecraft.client.Minecraft
import net.minecraft.client.renderer.block.model.ModelResourceLocation
import net.minecraftforge.fml.common.FMLCommonHandler
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly
import szewek.fl.network.FLNetUtil
import szewek.fl.network.FLNetUtilClient
import szewek.fl.util.PreRegister
import java.util.*

@SideOnly(Side.CLIENT)
class FLProxyClient : FLProxy() {
    private val prl = ArrayList<PreRegister>()

    override val netUtil: FLNetUtil
        get() = if (FMLCommonHandler.instance().side == Side.CLIENT) FLNetUtilClient.THIS else super.netUtil

    override fun addPreRegister(pr: PreRegister) {
        prl.add(pr)
    }

    override fun init() {
        val imm = Minecraft.getMinecraft().renderItem.itemModelMesher
        for (pr in prl) {
            val iset = pr.getItems()
            for (i in iset) {
                val mrl = ModelResourceLocation(i.registryName!!, "inventory")
                imm.register(i, 0, mrl)
            }
        }
        prl.clear()
    }
}
